src/collective/workspace/workspace.py

```diff
@@ -122,3 +137,58 @@ def remove_from_team(self, user):
         if membership is not None:
             membership.remove_from_team()
         return membership
+
+
+@adapter(IHasWorkspace, IObjectAddedEvent)
+def handle_workspace_added(context, event):
+    workspace = IWorkspace(context)
+    workspace_groups = get_workspace_groups_plugin(context)
+    for group_name in workspace.available_groups:
+        group_id = '{}:{}'.format(group_name.encode('utf8'), context.UID())
+        workspace_groups.addGroup(
+            group_id,
+            title='{}: {}'.format(group_name.encode('utf8'), context.Title()),
+        )
+
```

Comment:
This conflicts with ploneintranet, where we have an event handler for `IObjectAddedEvent` as well, and our more specific `IWorkspaceFolder`, so that our event handler gets called first. It does the following:
```
def workspace_added(ob, event):
    """
    when a workspace is created, we add the creator to
    the admin group. We then setup our placeful workflow

    """
    # Whoever creates the workspace should be added as an Admin
    userid = api.user.get_current().id
    ob.setCreators([userid])
    IWorkspace(ob).add_to_team(
        user=userid,
        groups=set(['Admins']),
    )
```
That means by the time the above `handle_workspace_added` event handler gets called, the groups 'Admins:ws_UID' and 'Members:ws_UID' already exist, leading to
_KeyError, 'Duplicate group ID: Admins:ws_UID'_

So I think a check like
```
if group_id not in workspace_groups._groups:
    workspace_groups.addGroup(
        ...
```
would be required.